src/data_source/storage/sql.rs

```diff
@@ -71,15 +78,30 @@ use self::{migrate::Migrator, transaction::PoolMetrics};
 ///
 /// As an example, this is the invocation used to load the default migrations from the
 /// `hotshot-query-service` crate. The migrations are located in a directory called `migrations` at
-/// the root of the crate.
+/// - PostgreSQL migrations are in `/migrations/postgres`.
+/// - SQLite migrations are in `/migrations/sqlite`.
 ///
 /// ```
 /// # use hotshot_query_service::data_source::sql::{include_migrations, Migration};
-/// let mut migrations: Vec<Migration> =
-///     include_migrations!("$CARGO_MANIFEST_DIR/migrations").collect();
-/// migrations.sort();
-/// assert_eq!(migrations[0].version(), 10);
-/// assert_eq!(migrations[0].name(), "init_schema");
+/// // For PostgreSQL
+/// #[cfg(not(feature = "embedded-db"))]
+/// {   
+///     let mut postgres_migrations: Vec<Migration> =
+///     include_migrations!("$CARGO_MANIFEST_DIR/migrations/postgres").collect();
+///     postgres_migrations.sort();
+///     assert_eq!(postgres_migrations[0].version(), 10);
+///     assert_eq!(postgres_migrations[0].name(), "init_schema");
+/// }
+///
+/// // For SQLite
+/// #[cfg(feature = "embedded-db")]
+/// {
+///     let mut sqlite_migrations: Vec<Migration> =
+///     include_migrations!("$CARGO_MANIFEST_DIR/migrations/sqlite").collect();
+///     sqlite_migrations.sort();
+///     assert_eq!(sqlite_migrations[0].version(), 10);
+///     assert_eq!(sqlite_migrations[0].name(), "init_schema");
+/// }
```

Comment:
Nit: this could be condensed
```rust
// For PostgreSQL
#[cfg(not(feature = "embedded-db"))]
let mut migrations: Vec<Migration> = include_migrations!("$CARGO_MANIFEST_DIR/migrations/postgres").collect();

// For SQLite
#[cfg(feature = "embedded-db")]
let mut migrations: Vec<Migration> = include_migrations!("$CARGO_MANIFEST_DIR/migrations/sqlite").collect();

migrations.sort();
assert_eq!(migrations[0].version(), 10);
assert_eq!(migrations[0].name(), "init_schema");
```